proc/process.c

```diff
@@ -395,26 +395,186 @@ static void process_tlsAssign(hal_tls_t *process_tls, hal_tls_t *tls, ptr_t tbss
 }
 
 
+static int process_validatePtr(void *map_start, size_t map_length, void *ptr_start, size_t ptr_size)
+{
+	void *map_end = ((char *)map_start) + map_length;
+	void *ptr_end = ((char *)ptr_start) + ptr_size;
+
+	return ((ptr_size == 0) ||
```

Comment:
Take a look here, this is two-person team researched optimal overlap check algorithm:
https://github.com/phoenix-rtos/plo/blob/486d7526322c6d835434d0b567dbc908c5522b35/syspage.c#L223

Reply:
So:
`((ptr_end > map_start) && (ptr_start < map_end) && (ptr_start < ptr_end)))`
should be enough

Reply:
but here it's not an overlap check but a check if the map includes the pointer